backends/kv/writer.go

```diff
@@ -685,3 +684,32 @@ func valueToTypedExpressionString(value interface{}) string {
 		return fmt.Sprintf("%v", value)
 	}
 }
+
+func valueToKeyString(value interface{}) string {
+	switch typedVal := value.(type) {
+	case string:
+		return typedVal
+	case time.Time:
+		seconds := typedVal.Unix()
+		nanos := typedVal.Nanosecond()
+		return fmt.Sprintf("%v:%v", seconds, nanos)
+	case bool:
+		// Python/storey compatibility by capitalizing
+		// IG-22141
+		if typedVal {
+			return "True"
+		}
+		return "False"
+	case float64, float32:
+		// Python/storey compatibility by deleting trailing zeros (except one trailing zero after period)
+		// IG-22140
+		str := fmt.Sprintf("%f", typedVal)
```

Comment:
Could be just str="{:.1f}".format(typedVal) if typedVal.is_integer() else str(typedVal)

Reply:
This is Go code, not python...

Reply:
Well, let me know if you have a suggestion in Go :)

Reply:
str := fmt.Sprintf("%.1f", typedVal) if typedVal == float64(int64(typedVal)) else strconv.FormatFloat(typedVal, 'f', -1, 64)

Just for fun :)